### PR TITLE
trace: fix feature flags nonsense

### PR DESCRIPTION
**Pull request by hawkw**
It turns out that the reason ANSI colors were on is because cargo
dev-dependencies don't work the way we thought they did — feature flags
from dev-dependencies are unified with features from non-dev
dependencies, so the ANSI feature flag was being enabled.

There's a magical setting that makes cargo stop gaslighting me, but we
have to update our Rust toolchain to use it. This branch does that.

Also, I moved all the test tracing configuration stuff into the
`linkerd-tracing` crate, which is maybe a little bit nicer.

### CODE REVIEW

File: linkerd/app/inbound/Cargo.toml [hunk 1]
```diff
@@ -32,4 +32,4 @@ linkerd-app-test = { path = "../test" }
 linkerd-io = { path = "../../io", features = ["tokio-test"] }
 tokio = { version = "1", features = ["full", "macros"]}
 tokio-test = "0.4"
-tracing-subscriber = "0.2.16"
+linkerd-tracing = { path = "../../tracing", features = ["ansi"] }
```
Comment (olix0r) [inline linkerd/app/inbound/Cargo.toml:35]: https://marketplace.visualstudio.com/items?itemName=samverschueren.final-newline

File: linkerd/app/outbound/Cargo.toml [hunk 2]
```diff
@@ -35,4 +35,4 @@ linkerd-app-test = { path = "../test" }
 linkerd-io = { path = "../../io", features = ["tokio-test"] }
 tokio = { version = "1", features = ["full", "macros"]}
 tokio-test = "0.4"
-tracing-subscriber = "0.2.16"
+linkerd-tracing = { path = "../../tracing", features = ["ansi"] }
```
Comment (olix0r) [inline linkerd/app/outbound/Cargo.toml:38]:
```suggestion
linkerd-tracing = { path = "../../tracing", features = ["ansi"] }
```

File: linkerd/cache/Cargo.toml [hunk 3]
```diff
@@ -17,4 +17,4 @@ tracing = "0.1.23"
 
 [dev-dependencies]
 tokio = { version = "1", default-features = false, features = ["macros", "test-util", "time"] }
-tracing-subscriber = "0.2.16"
+linkerd-tracing = { path = "../tracing", features = ["ansi"] }
```
Comment (olix0r) [inline linkerd/cache/Cargo.toml:20]:
```suggestion
linkerd-tracing = { path = "../tracing", features = ["ansi"] }
```

File: linkerd/cache/src/lib.rs [hunk 4]
```diff
@@ -207,9 +207,8 @@ where
 }
 
 #[cfg(test)]
-#[tokio::test]
+#[tokio::test(flavor = "current_thread")]
 async fn test_idle_retain() {
-    let _ = tracing_subscriber::fmt::try_init();
```
Comment (olix0r) [inline linkerd/cache/src/lib.rs:212]: should tracing be enabled here?

File: linkerd/proxy/http/Cargo.toml [hunk 5]
```diff
@@ -45,4 +45,4 @@ tokio-test = "0.4"
 
 [dev-dependencies]
 tokio-test = "0.4"
-tracing-subscriber = "0.2.16"
+linkerd-tracing = { path = "../../tracing", features = ["ansi"] }
```
Comment (olix0r) [inline linkerd/proxy/http/Cargo.toml:48]:
```suggestion
linkerd-tracing = { path = "../../tracing", features = ["ansi"] }
```

File: linkerd/tls/Cargo.toml [hunk 6]
```diff
@@ -28,6 +28,6 @@ untrusted = "0.7"
 [dev-dependencies]
 linkerd-identity = { path = "../identity", features = ["test-util"] }
 linkerd-proxy-transport = { path = "../proxy/transport" }
+linkerd-tracing = { path = "../tracing", features = ["ansi"]}
 tokio = { version = "1", features = ["rt-multi-thread"] }
-tower = { version = "0.4.7", default-features = false, features = ["util"] }
-tracing-subscriber = "0.2.16"
+tower = { version = "0.4.7", default-features = false, features = ["util"] }
```
Comment (olix0r) [inline linkerd/tls/Cargo.toml:33]:
```suggestion
tower = { version = "0.4.7", default-features = false, features = ["util"] }
```

**Review by olix0r — APPROVED** [review]
pending CI fixes

**Comment by olix0r** [comment]
@hawkw we're going to abandon this branch in favor of boxing/upgrading right?

**Comment by hawkw** [comment]
> @hawkw we're going to abandon this branch in favor of boxing/upgrading right?

no, this _is_ the boxing/upgrading branch :) we already merged the change that just moved the feature flags around

**Comment by hawkw** [comment]
@olix0r btw, i've updated this with your recent changes from `main`.

**Pull request closed without merging**